File: Binary tree.cpp
#include <bits/stdc++.h>
using namespace std;

struct treeNode{
	int data;
	treeNode *left, *right;
	treeNode(int data){
		this->data=data;
		left=right=NULL;
	}
};

void postOrder(treeNode *current_node){
	if (current_node==NULL) return;
	postOrder(current_node->left);
	postOrder(current_node->right);
	cout<<current_node->data<<" ";
}

void preOrder(treeNode *current_node){
	if(current_node==NULL) return;
	cout<<current_node->data<<" ";
	preOrder(current_node->left);
	preOrder(current_node->right);
}

void inOrder(treeNode *current_node){
	if(current_node==NULL) return;
	inOrder(current_node->left);
	cout<<current_node->data<<" ";
	inOrder(current_node->right);
}

int main(){
	treeNode *root=new treeNode(1);
	root->left=new treeNode(2);
	root->right=new treeNode(3);
	root->left->right=new treeNode(4);
	root->left->left=new treeNode(5);
	postOrder(root);
	cout<<endl;
	preOrder(root);
	cout<<endl;
	inOrder(root);
	return 0;
}

File: Python Implementation/BinaryTree.py
class Node:
  def __init__(self, val):
    self.data = val 
    self.left = None 
    self.right = None

def inorder(root):
  if root is None: 
    return
  inorder(root.left)
  print(root.data)
  inorder(root.right)

def preorder(root):
  if root is None: 
    return
  print(root.data)
  preorder(root.left)
  preorder(root.right)

def postorder(root):
  if root is None: 
    return
  postorder(root.left)
  postorder(root.right)
  print(root.data)

root = Node(12)
root.left = Node(11)
root.right = Node(20)
root.left.left = Node(30)
root.left.right = Node(10)
root.right.left = Node(0)
root.left.left.right = Node(9)

inorder(root)
print('-'*10)
preorder(root)
print('-'*10)
postorder(root)
print('-'*10)

'''
30
9
11
10
12
0
20
----------
12
11
30
9
10
20
0
----------
9
30
10
11
0
20
12
----------
'''
